Add size and align attributes to xml output

iMichka opened the pull request:
This is expected by pygccxml, at least for the first batch of tests I am currently fixing.

Maybe align and size attributes will need to be added to other declarations later, I did only cover these two (RecordDecl and BuiltinType) for the moment.

bradking commented:
Thanks for working on this.  I originally left out these attributes intentionally because they never really belonged in gccxml in the first place since it didn't try to match the architecture.  However, IIRC in CastXML the underlying Clang is configured for the correct target architecture, so we can actually depend on the size and alignment values.

Is doing this for just `RecordDecl` and `BulitinType` sufficient for pygccxml's tests, or will more be needed?

iMichka commented:
I don't know if it will be needed at other places. I am still fighting with the first errors, but it seems that would cover most of it as the tests are looking only those attributes for the two cases here. If you want to wait a little bit it's no problem so that I have a clearer picture of what is needed.

bradking commented:
Okay, thanks.  I'd prefer these attributes be added everywhere possible at once anyway.

iMichka commented:
Rebased on master. I added the offset attribute to the ```<Field>``` tag; which was also expected by pygccxml.

One thing which is missing is the ```mangled/demangled``` attributes. I need to check how to get those from the clang AST, which seems not to be straightforward. Maybe you have an idea for these two ?

bradking commented:
Let's defer mangling information for now.

Please split the `Field` `offset` support into a separate commit, perhaps preceding this one.

bradking commented:
Also, several test output expectation files need to be updated for tests that only run on specific platforms.  See `float128` for example.

iMichka commented:
Okay, I think I did everything :)

- Rebased on master
- Split the commit in 2
- Removed the to string conversion
- Introduced a single PrintABIAttributes method for size and align
- Used size="[0-9]+" align="[0-9]+" in the test files
- Added size and align for float128, and made sure all FundamentalTypes have size and align
- Put the dynamic casting in an if statement, as per llvm manual (http://llvm.org/docs/ProgrammersManual.html#the-isa-cast-and-dyn-cast-templates), if that is what you ment by catching a null pointer.

bradking commented:
Great, thanks.  I've made minor tweaks and integreated into `master` as commit 57c93dfba8395299cd939c40dd894f40cfd4ec28 and commit 45d5a24874afa9343df9048bb240b655e9b49965.

iMichka commented:
Thanks
